src/daft-connect/src/convert.rs

```diff
@@ -0,0 +1,45 @@
+mod data_conversion;
+mod expression;
+mod formatting;
+mod plan_conversion;
+mod schema_conversion;
+
+use std::{collections::HashMap, pin::Pin, sync::Arc};
+
+use common_daft_config::DaftExecutionConfig;
+use common_error::{DaftError, DaftResult};
+use daft_logical_plan::LogicalPlanRef;
+use daft_table::Table;
+pub use data_conversion::convert_data;
+use futures::{stream, Stream, StreamExt};
+pub use schema_conversion::connect_schema;
+
+pub fn run_local(
+    logical_plan: &LogicalPlanRef,
+) -> DaftResult<impl Stream<Item = DaftResult<Table>>> {
+    let physical_plan = daft_local_plan::translate(logical_plan)?;
+    let cfg = Arc::new(DaftExecutionConfig::default());
+    let psets = HashMap::new();
+
+    let stream = daft_local_execution::run_local(&physical_plan, psets, cfg, None)?;
+
+    let stream = stream
+        .map(|partition| match partition {
+            Ok(partition) => partition.get_tables().map_err(DaftError::from),
+            Err(err) => Err(err),
+        })
+        .flat_map(|tables| match tables {
+            Ok(tables) => {
+                let tables = Arc::try_unwrap(tables).unwrap();
+
+                let tables = tables.into_iter().map(Ok);
+                let stream: Pin<Box<dyn Stream<Item = DaftResult<Table>>>> =
+                    Box::pin(stream::iter(tables));
+
+                stream
+            }
+            Err(err) => Box::pin(stream::once(async { Err(err) })),
+        });
+
+    Ok(stream)
+}
```

Comment:
can we move this into `daft_local_execution`. I think this'll be helpful for my sql work. 

also nit: the name `run_local` is imo pretty bad _(I know the original one is also called `run_local`)_. Could we rename this to something more descriptive like `execute_logical_plan`

Reply:
actually going to save this for another PR as it looks like this isn't actually used anywhere yet for ranges